contracts/Rules.sol

```diff
@@ -39,61 +28,29 @@ contract Rules is AdminProxy, RulesProxy {
     // head of linked list
     bytes headWhitelist;
 
-    // AUTHORIZATION
-    constructor (address ingressAddress) public {
-        // add the deploying contract address as first admin
-        Admin memory orig = Admin(msg.sender, true);
-        admins[msg.sender] = orig;
-        adminKeys.push(msg.sender);
-        adminCount = adminCount + 1;
-
-        // Record the Ingress contract's address
-        ingressContractAddress = ingressAddress;
-    }
+    address private ingressContractAddress;
+    address private adminContractAddress;
 
-    // AUTHORIZATION: LIST OF ADMINS
-    modifier onlyAdmin()
-    {
-        require(
-            admins[msg.sender].adminAddress != address(0),
-            "Sender not authorized."
-        );
-        require(admins[msg.sender].active == true, "Sender not authorized");
+    modifier onlyOnEditMode() {
+        require(!readOnlyMode, "In read only mode: rules cannot be modified");
         _;
     }
 
-    function isAuthorized(address source) public view returns (bool) {
-        return admins[source].active;
-    }
-
-    function addAdmin(address _newAdmin) public onlyAdmin returns (bool) {
-        if (admins[_newAdmin].active) {
-            return false;
-        }
-        adminKeys.push(_newAdmin);
-        Admin memory newAdmin = Admin(_newAdmin, true);
-        admins[_newAdmin] = newAdmin;
-        adminCount = adminCount + 1;
-        return true;
+    modifier onlyAdmin() {
+        require(adminContractAddress != address(0), "Ingress contract must have Admin contract registered");
+        require(Admin(adminContractAddress).isAuthorized(msg.sender), "Sender not authorized");
+        _;
     }
 
-    function removeAdmin(address oldAdmin) public onlyAdmin returns (bool) {
-        admins[oldAdmin].active = false;
-        adminCount = adminCount - 1;
-        return true;
+    constructor (address ingressAddress) public {
+        ingressContractAddress = ingressAddress;
+        Ingress ingressContract = Ingress(ingressContractAddress);
+        adminContractAddress = ingressContract.getContractAddress(ingressContract.ADMIN_CONTRACT());
```

Comment:
This needs to be looked up each time, not set at construction. We've got no reliance that the rules contract will only be installed after the Admin and that the Rules will be updated every time the Admin will be.